shared/src/main/scala/LiterateLexer.scala

```diff
@@ -0,0 +1,158 @@
+package vyxal
+
+import java.util.regex.Pattern
+import scala.util.matching.Regex
+import scala.util.parsing.combinator.*
+import LiterateToken.*
+
+val hardcodedKeywords = Map(
+  // These can't go in the big map, because that's autogenerated
+  "if" -> "[",
+  "endif" -> "}",
+  "end-if" -> "}",
+  "for" -> "(",
+  "endfor" -> "}",
+  "end-for" -> "}",
+  "while" -> "{",
+  "endwhile" -> "}",
+  "end-while" -> "}",
+  "lambda" -> "λ",
+  "endlambda" -> "}",
+  "end-lambda" -> "}",
+  "end" -> "}",
+  "else" -> "|",
+  "do-to-each" -> "(",
+  "body" -> "|",
+  "branch" -> "|",
+  "close-all" -> "]"
+)
+
+enum LiterateToken(val value: Object):
+  // Object instead of String like the normal lexer because it's way easier
+  case Word(override val value: String) extends LiterateToken(value)
+  case AlreadyCode(override val value: String) extends LiterateToken(value)
+  // This is for strings that are already in SBCS form
+  case LitComment(override val value: String) extends LiterateToken(value)
+  case LambdaBlock(override val value: List[Object])
+      extends LiterateToken(value.toString)
+
+  case ListToken(override val value: List[Object])
+      extends LiterateToken(value.toString)
```

Comment:
Why not `LiterateToken(value)`? `value.toString` will look like `List(..., ..., ...)`

Reply:
Copilot!